### PR TITLE
chart: introduce v1-data-engine setting

### PR DESCRIPTION
**Pull request by derekbit**
longhorn/longhorn#7095

### CODE REVIEW

**Review by innobead — APPROVED** [review]
LGTM

**Comment by innobead** [comment]
@jillian-maroket Please help review the description of the new settings. After that, we can merge this PR.

File: chart/questions.yaml [hunk 1]
```diff
@@ -861,3 +867,11 @@ questions:
       - "rke1"
       - "rke2"
       - "k3s"
+  - variable: defaultSettings.v2DataEngineGuaranteedInstanceManagerCPU
+    label: Guaranteed Instance Manager CPU for V2 Data Engine
+    description: "This integer value indicates how many millicpus on each node will be reserved for each instance manager Pod for v2 data engine. By default, the SPDK target daemon within an instance manager Pod utilizes 1 CPU core. Ensuring a minimum CPU usage is essential for sustaining engine and replica stability, especially during periods of high node workload."
```
Comment (jillian-maroket) [inline chart/questions.yaml:872]:
```suggestion
    description: "Number of millicpus on each node to be reserved for each instance manager pod for the V2 Data Engine. By default, the Storage Performance Development Kit (SPDK) target daemon within each instance manager pod uses 1 CPU core. Configuring a minimum CPU usage value is essential for maintaining engine and replica stability, especially during periods of high node workload. The default value is 1250 millicpus."
```

File: chart/values.yaml [hunk 2]
```diff
@@ -290,7 +290,7 @@ defaultSettings:
   # -- This interval in seconds determines how long Longhorn will wait before re-downloading the backing image file
   # when all disk files of this backing image become failed or unknown.
   backingImageRecoveryWaitInterval: ~
-  # -- This integer value indicates how many percentages of the total allocatable CPU on each node will be reserved for each instance manager Pod.
+  # -- This integer value indicates how many percent of the total allocatable CPU on each node will be reserved for each instance manager Pod for v1 data engine.
```
Comment (jillian-maroket) [inline chart/values.yaml:293]:
```suggestion
  # -- Percentage of the total allocatable CPU resources on each node to be reserved for each instance manager pod for the V1 Data Engine.
```

File: chart/values.yaml [hunk 3]
```diff
@@ -328,11 +328,17 @@ defaultSettings:
   backupConcurrentLimit: ~
   # -- This setting controls how many worker threads per restore concurrently.
   restoreConcurrentLimit: ~
+  # -- This allows users to activate v1 data engine.
```
Comment (jillian-maroket) [inline chart/values.yaml:331]:
```suggestion
  # -- Setting that allows you to enable the V1 Data Engine.
```

File: chart/values.yaml [hunk 4]
```diff
@@ -290,7 +290,7 @@ defaultSettings:
   # -- This interval in seconds determines how long Longhorn will wait before re-downloading the backing image file
   # when all disk files of this backing image become failed or unknown.
   backingImageRecoveryWaitInterval: ~
-  # -- This integer value indicates how many percentages of the total allocatable CPU on each node will be reserved for each instance manager Pod.
+  # -- This integer value indicates how many percent of the total allocatable CPU on each node will be reserved for each instance manager Pod for v1 data engine.
   # You can leave it with the default value, which is 12%.
```
Comment (jillian-maroket) [inline chart/values.yaml:294]:
```suggestion
  # The default value is 12%.
```

File: chart/values.yaml [hunk 5]
```diff
@@ -328,11 +328,17 @@ defaultSettings:
...
   # -- This setting controls how many worker threads per restore concurrently.
   restoreConcurrentLimit: ~
+  # -- This allows users to activate v1 data engine.
+  v1DataEngine: ~
   # -- This allows users to activate v2 data engine based on SPDK.
   # Currently, it is in the preview phase and should not be utilized in a production environment.
   v2DataEngine: ~
   # -- This setting allows users to enable the offline replica rebuilding for volumes using v2 data engine.
   offlineReplicaRebuilding: ~
+  # -- This integer value indicates how many millicpus of CPU on each node will be reserved for each instance manager Pod for v2 data engine.
```
Comment (jillian-maroket) [inline chart/values.yaml:338]:
```suggestion
  # -- Number of millicpus on each node to be reserved for each instance manager pod for the V2 Data Engine.
```

File: chart/values.yaml [hunk 6]
```diff
@@ -328,11 +328,17 @@ defaultSettings:
...
+  # You can leave it with the default value, which is 1250 millicpus.
```
Comment (jillian-maroket) [inline chart/values.yaml:339]:
```suggestion
  # The default value is 1250 millicpus.
```

File: chart/values.yaml [hunk 7]
```diff
@@ -328,11 +328,17 @@ defaultSettings:
...
+  v2DataEngineGuaranteedInstanceManagerCPU: ~
+  # -- Enabling this setting will notify Longhorn that the cluster is using Kubernetes Cluster Autoscaler.
```
Comment (jillian-maroket) [inline chart/values.yaml:341]:
```suggestion
  # -- Setting that notifies Longhorn that the cluster is using the Kubernetes Cluster Autoscaler.
```

**Review by jillian-maroket — COMMENTED** [review]
Let me know if you have concerns about the changes.

**Comment by innobead** [comment]
@derekbit Let's update/correct the description of the settings in other places (doc, longhorn-manager?) after merging.

**Comment by derekbit** [comment]
> @derekbit Let's update/correct the description of the settings in other places (doc, longhorn-manager?) after merging.

@jillian-maroket @innobead 
All are updated. Thank you.

File: chart/values.yaml [hunk 8]
```diff
@@ -328,11 +328,17 @@ defaultSettings:
...
+  v1DataEngine: ~
+  # -- This allows users to activate v2 data engine based on SPDK. By default, false.
   # Currently, it is in the preview phase and should not be utilized in a production environment.
   v2DataEngine: ~
   # -- This setting allows users to enable the offline replica rebuilding for volumes using v2 data engine.
   offlineReplicaRebuilding: ~
+  # -- Number of millicpus on each node to be reserved for each instance manager pod when the V2 Data Engine is enabled.
+  # The default value is 1250 millicpus.
+  v2DataEngineGuaranteedInstanceManagerCPU: ~
+  # -- Setting that notifies Longhorn that the cluster is using the Kubernetes Cluster Autoscaler.
```
Comment (innobead) [inline chart/values.yaml:341]: Why do we have this comment next here? It should be for the cluster autoscaler setting.

**Review by innobead — CHANGES_REQUESTED** [review]
LGTM, just one comment.

File: chart/values.yaml (hunk 8 above)
Comment (derekbit) [reply chart/values.yaml:341]: @jillian-maroket updated the description as well, so update it together.

**Review by innobead — APPROVED** [review]

**Pull request merged**